src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -105,10 +106,37 @@ def metadata_call_credentials(metadata_plugin, name=None):
     A CallCredentials object for use in a GRPCCallOptions object.
   """
   if name is None:
-    name = metadata_plugin.__name__
```

Comment:
By default python objects don't have a __name__ attribute.  If I pass in an arbitrary MetadataPlugin, it will throw an exception.